src/api-service/__app__/timer_workers/__init__.py

```diff
@@ -42,13 +43,17 @@ def main(mytimer: func.TimerRequest, dashboard: func.Out[str]) -> None:  # noqa:
             autoscale_pool(pool)
 
     Node.mark_outdated_nodes()
+    # ensure nodes are updated in a consistent order, which is needed for
+    # autoscaling of pools
```

Comment:
Could you elaborate on this just a bit? I take this to mean "consistent across timer-triggered invocations", but it would be helpful to explain why, and how it interacts with auto-scaling in particular.